### PR TITLE
Refactoring to extract a definition

### PR DESCRIPTION
**Pull request by mainej**
This is the basic version of implementing #1136. It does not do find-and-replace.

- [x] I created an issue to discuss the problem I am trying to solve or an open issue already exists.
- [x] I added a new entry to [CHANGELOG.md](https://github.com/clojure-lsp/clojure-lsp/blob/master/CHANGELOG.md)
- [x] I updated documentation if applicable (`docs` folder)

### CODE REVIEW

File: lib/src/clojure_lsp/feature/code_actions.clj [hunk 1]
```diff
@@ -155,6 +155,13 @@
              :command   "extract-function"
              :arguments [uri line character "new-function"]}})
 
+(defn ^:private extract-definition-action [uri line character]
+  {:title   "Extract definition"
```
Comment (ericdallo) [inline lib/src/clojure_lsp/feature/code_actions.clj:159]: WDYT about `extract do def`? seems more clear to me what this code action will do

File: lib/src/clojure_lsp/refactor/transform.clj [hunk 2]
```diff
@@ -453,6 +453,32 @@
...
+  (boolean (extract-definition-params zloc)))
+
+(defn extract-definition [zloc def-name]
+  (when-let [{:keys [zloc form-loc]} (extract-definition-params zloc)]
+    (let [expr-node (z/node zloc)
+          expr-meta (meta expr-node)
+
+          def-name (or def-name "new-value")
+          expr-edit (z/of-node (symbol def-name))
+          def-loc (-> (format "(def ^:private %s\n  )" def-name)
```
Comment (ericdallo) [inline lib/src/clojure_lsp/refactor/transform.clj:475]: IIRC extract function checks for the `:use-metadata-for-privacy?` setting which is actually disable by default, I think we should check of that and use `def-` if not enabled instead

File: lib/src/clojure_lsp/feature/code_actions.clj (hunk 1 above)
Comment (mainej) [reply lib/src/clojure_lsp/feature/code_actions.clj:159]: Sounds good—that is much clearer.

File: lib/src/clojure_lsp/refactor/transform.clj (hunk 2 above)
Comment (mainej) [reply lib/src/clojure_lsp/refactor/transform.clj:475]: I thought that too at first, but then I remembered—there's no such thing as `def-`!

File: lib/src/clojure_lsp/feature/code_actions.clj (hunk 1 above)
Comment (mainej) [reply lib/src/clojure_lsp/feature/code_actions.clj]: Done.

File: lib/src/clojure_lsp/refactor/transform.clj (hunk 2 above)
Comment (ericdallo) [reply lib/src/clojure_lsp/refactor/transform.clj:475]: Oh, great point 😂

**Review by ericdallo — APPROVED** [review]
@mainej Feel free to solve conflicts and merge, thanks!

**Pull request merged**